source/develop/release-management/features/virt/dedicated-cpu.md

```diff
@@ -0,0 +1,202 @@
+be used by any
+other VM because of restrictions requirements on the threads of the core. And
+finally there is the set of CPUs that are not strictly assigned to any
+particular VM and are shared between all the VMs with *shared* policy. This
+subset is hereby referred to as *shared pool*. When scheduling VMs engine also
+takes into consideration use of host CPUs that play a special role, e.g. CPU
+that VDSM is pinned to or CPUs reserved to host on SAP HANA systems.
+
+The available CPU policies are:
+
+  * *shared*: *(default)* The guest vCPUs will be allowed to roam among all
+    pCPUs from shared pool.
+  * *dedicated*: The guest vCPUs will be strictly pinned to a set of host pCPUs
+    (similarly to static CPU pinning). The set of pCPUs will be chosen to match
+    the required guest CPU topology. If the host has SMT architecture, thread
+    siblings are preferred.
+  * *isolate-threads*: If the host does not have SMT architecture this is same
+    as *dedicated* policy. If the host has SMT architecture each vCPU is placed
+    on a different physical core and all thread siblings are excluded from use
+    by this and any other VM (rendering all but one thread from a core unusable).
+  * *require-threads*: The host must have an SMT architecture. Each vCPU is
+    allocated on thread siblings and threads from single core can only be
+    assigned to this VM. If some threads remain unused these will be excluded
+    from use by any other VM.
+
+
+
+
+## Prerequisites
+
+There are no special prerequisites for this feature.
+
+
+## Security
+
+This feature is primarily designed for performance tuning and is not meant to
+be a security feature. It is possible for other VMs to share the dedicated CPUs
+for a short period of time when the VM is brought up (boot, after migration,
+restore from snapshots, etc.). This is not so much problem during boot because
+by the time OS is started the CPUs should be already evicted, but it could
+potentially provide a window of opportunity (for exploiting CPU
+vulnerabilities) after finished migration or when restoring VM from snapshot.
+
+
+## Limitations
+
+See the *Security* section.
+
+Currently, when a CPU is already dedicated to some VM, VDSM will not move the
+VM onto another CPU to make it free to another VM. This may be addressed in
+the future (see *Future Work* section), but for now this means:
+
+  * It may be impossible to start VMs with CPU pinning on that host because
+    the requested CPUs are already occupied. It is suggested to start VMs with
+    CPU pinning before any VMs with dedicated CPU policy.
+
+  * It may be impossible to start a VM with dedicated CPUs even though there
+    seems to be enough space in terms of number of used vs. free CPUs.
+
+
+## Changes to Engine and Vdsm
+
+The high level overview of the workflow in Engine and Vdsm is following:
+
+1.  When host switches to *UP* state engine queries it to find out which pCPUs
```

Comment:
before the host reaches the UP state, there should be no VMs running on it (unless it was non-responding) - did you mean a different state?